src/common/printer-helpers.js

```diff
@@ -8,7 +8,7 @@ import {
 const { group, indent, join, line, softline, hardline } = doc.builders;
 
 export const printComments = (node, path, options, filter = () => true) => {
-  if (!node.comments) return '';
+  if (!node.comments) return [];
```

Comment:
if this functions always returns an array, then we can always ask for `comments.length` instead of `comments?.length`